static/js/components/dashboard/CourseDescription_test.js

```diff
@@ -50,6 +53,84 @@ describe('CourseDescription', () => {
     }
   });
 
+  it('does not show view the course on edX link if enrollment_url is no define', () => {
+    const NOT_ENROLLED_STATUSES =[
+      STATUS_WILL_ATTEND, STATUS_OFFERED, STATUS_PENDING_ENROLLMENT
+    ];
+
+    for (let status of NOT_ENROLLED_STATUSES) {
+      let course = findAndCloneCourse(course => (
+        course.runs.length > 0 &&
+        course.runs[0].status === status
+      ));
+      let firstRun = course.runs[0];
+      const wrapper = shallow(<CourseDescription courseRun={firstRun} courseTitle={course.title} />);
+      let elements = getElements(wrapper);
+
+      assert.lengthOf(elements.titleLink, 0);
+    }
+  });
+
+  it('view the course on edX link when course start date is past chages', () => {
+    const EXPECTED_STATUSES = [
+      STATUS_CAN_UPGRADE, STATUS_MISSED_DEADLINE
+    ];
+
+    for (let status of EXPECTED_STATUSES) {
+      let course = findAndCloneCourse(course => (
+        course.runs.length > 0 &&
+        course.runs[0].status === status
+      ));
+      let firstRun = course.runs[0];
+      firstRun.course_start_date = moment().subtract(2, 'days').format();
+      firstRun.enrollment_url = null;
+      const wrapper = shallow(<CourseDescription courseRun={firstRun} courseTitle={course.title} />);
+      let elements = getElements(wrapper);
+
+      assert.equal(elements.titleLink.text(), 'View on edX');
+      assert.isAbove(elements.titleLink.props().href.length, 0);
+    }
+  });
+
+  it('view the course on edX link when course is not start', () => {
+    const EXPECTED_STATUSES = [
+      STATUS_CAN_UPGRADE, STATUS_MISSED_DEADLINE
+    ];
+
+    for (let status of EXPECTED_STATUSES) {
+      let course = findAndCloneCourse(course => (
+        course.runs.length > 0 &&
+        course.runs[0].status === status
+      ));
+      let firstRun = course.runs[0];
+      firstRun.course_start_date = moment().add(2, 'days').format();
+      firstRun.enrollment_url = 'http://example.com';
+      const wrapper = shallow(<CourseDescription courseRun={firstRun} courseTitle={course.title} />);
+      let elements = getElements(wrapper);
+      assert.equal(elements.titleLink.text(), 'View on edX');
+      assert.isAbove(elements.titleLink.props().href.length, 0);
+    }
+  });
+
+  it('view the course on edX link when course is not start and enrollment_url not set', () => {
+    const EXPECTED_STATUSES = [
+      STATUS_CAN_UPGRADE, STATUS_MISSED_DEADLINE
+    ];
+
+    for (let status of EXPECTED_STATUSES) {
+      let course = findAndCloneCourse(course => (
+        course.runs.length > 0 &&
+        course.runs[0].status === status
+      ));
+      let firstRun = course.runs[0];
+      firstRun.course_start_date = moment().add(2, 'days').format();
+      firstRun.enrollment_url = null;
+      const wrapper = shallow(<CourseDescription courseRun={firstRun} courseTitle={course.title} />);
+      let elements = getElements(wrapper);
+      assert.lengthOf(elements.titleLink, 0);
+    }
+  });
+
```

Comment:
Here's a rough list of the cases that we need to test for this change:

1. [Current/past enroll statuses](https://github.com/mitodl/micromasters/pull/2073/files#diff-a706cfee3d180faef170a50939f2ab11R72) yield a `course_id` link **(you don't have this yet)**
1. "Can upgrade"/"missed deadline" statuses with a course run that has started yields a `course_id` link **(you have this, but you don't test for the presence of `course_id` in the link)**
1. "Can upgrade"/"missed deadline" statuses with course run that has not yet started yields an `enrollment_url` link **(you have this, but you don't test for the presence of `enrollment_url` in the link)**
1. "Can upgrade"/"missed deadline" statuses with null course start date yield an `enrollment_url` link *(you do not have this yet)**
1. Course run with a non-enroll status yields an `enrollment_url` link **(you do not have this yet)**
1. Course run with a non-enroll status with null enrollment_url results in no link (you have this one)
1. Current/past enroll statuses with null course_id results in no link **(you do not have this yet, and this depends on whether or not we want this case to link to the base edX url)**

in addition to those, the test case with the description 'does not show a link to view the course on edX if the course run lacks an id' (line 134 right now) is passing, but it's incorrect. The link in that case is not showing because the course run's status is `STATUS_OFFERED` and there is no `enrollment_url`. it has nothing to do with the null `course_id`